dependency/vault_read.go

```diff
@@ -208,6 +208,16 @@ func shimKVv2Path(rawPath, mountPath string) string {
 		if strings.HasPrefix(p, "data/") || strings.HasPrefix(p, "metadata/") {
 			return rawPath
 		}
+
+		// If the raw path contains "/data/", it means the namespace on the Vault client
+		// is different from the namespace prefixing the rawPath. We want to
```

Comment:
Is this true? Would this be true for KVv1 secrets, too?

Reply:
Hmm, another concern: if I'm accessing KVv2 secrets without `/data/` today, then this function will automatically add it for me -- we should try to preserve that behaviour if possible, so that it works for all namespaces

Reply:
> Hmm, another concern: if I'm accessing KVv2 secrets without /data/ today, then this function will automatically add it for me -- we should try to preserve that behaviour if possible, so that it works for all namespaces

I believe this should still work - below there's an extra if statement that adds `"data/" `to the path if it doesn't exist, and that will still be preserved.

Reply:
Aha! Thank you for assuaging my concerns there!

I think as part of testing this (and before we merge this code ideally) we should triple check to make sure this works for secrets that rely on the bit of the code that adds `"data/"` when it's missing _and_ we have this namespace setup where it's in the client as well as the path. I'm a little concerned about that situation since we return here before we get to the bit we add `"data/"`.

Reply:
That makes sense, and I modified this to work correctly. I also have some vault-enterprise tests here that test this: https://github.com/hashicorp/vault-enterprise/pull/5811